qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/diagnostics/CollectHtmlInputNamesVisitor.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.diagnostics;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.redhat.qute.parser.html.scanner.HtmlScanner;
+import com.redhat.qute.parser.html.scanner.TokenType;
+import com.redhat.qute.parser.template.ASTVisitor;
+import com.redhat.qute.parser.template.Text;
+
+/**
+ * Collect input section names.
+ *
+ */
+public class CollectHtmlInputNamesVisitor extends ASTVisitor {
+
+	private final List<String> htmlInputNames;
+
+	public CollectHtmlInputNamesVisitor() {
+		this.htmlInputNames = new ArrayList<String>();
+
+	}
+
+	public List<String> getHtmlInputNames() {
+		return htmlInputNames;
+	}
+
+	@Override
+	public boolean visit(Text text) {
+		// scan with html scanner
+		HtmlScanner scanner = HtmlScanner
+				.createScanner(text.getOwnerTemplate().getText(), text.getStart(), text.getEnd());
+		TokenType token = scanner.scan();
+		String currentAttrName = null;
+		String currentElementName = null;
+		while (token != TokenType.EOS) {
+			switch (token) {
+				case ElementName:
+					currentElementName = scanner.getTokenText();
+					break;
+				case AttributeName:
+					if (currentElementName != null && "input".equals(currentElementName)) {
+						currentAttrName = scanner.getTokenText();
+					}
+					break;
+				case AttributeValue:
+					if (currentAttrName != null && "name".equals(currentAttrName)) {
+						StringBuilder inputName = new StringBuilder();
+						inputName.append(scanner.getTokenText().replace("\"", "").replace("\'", ""));
```

Comment:
Don't use `scanner.getTokenText()`, but loop for from `scanner.getTokenOffset()` to `scanner.getTokenEnd()` to collect attribut evalue character by character and remove the quote if needed because with your current code you create 3 instances of String (scanner.getTokenText() + replace + replace = 3).